ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultDiskResourceService.java

```diff
@@ -60,6 +60,10 @@ private boolean isStarted() {
   @Override
   public void start(final ServiceProvider<Service> serviceProvider) {
     innerStart(serviceProvider);
+    if (persistenceService != null && !persistenceService.isClean()) {
```

Comment:
We shouldn't need to null-check here. Persistence service should always be non-null because we depend on it.

Reply:
@chrisdennis 
Seven existing test cases are getting failed in DefaultDiskResourceServiceTest.java ("WithoutPersistenceService") if not handling the null check. Please share your thoughts on this. Thanks.

Reply:
You need to look at the `@ServiceDependency` annotation that is applied to this class, and think about how that annotation is interpreted by the ServiceLocator. Then note that the tests you're looking at are not using the ServiceLocator to provide the services to the DefaultDiskResourceService.